components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1029,10 +1064,25 @@ void RewardsServiceImpl::LoadLedgerState(
 
 void RewardsServiceImpl::OnLedgerStateLoaded(
     ledger::OnLoadCallback callback,
-    const std::string& data) {
+    std::pair<std::string, base::Value> state) {
   if (!Connected())
     return;
 
+  if (state.second.is_dict()) {
```

Comment:
I'd really prefer to encapsulate all of this logic in a P3A method. Maybe it could be a `friend` and we can pass in `this` so you have access to everything?

Reply:
same thing with https://github.com/brave/brave-core/pull/3242/files#diff-a5d16be9c9b52a6df7b866b1d75d3f5fR206

Reply:
this is not critical in any case and can be follow-up

Reply:
I believe we should not consider P3A code as something non-usual or alien. Collecting these values is a normal and routine part of the business logic, sometimes quite strongly tied to the actual code - Chromium is full of examples.
I'd say we should focus on better architecture and granularity of our classes, to make adding product analytics code easy.

Reply:
I'm not saying that it's non-usual or alien, it's just about code organization. RewardsServiceImpl is already severely bloated and we don't want to just keep dumping more code in it. It also makes rebase a nightmare because more changes affect the same files. We need to break the code up into smaller pieces to make it more manageable.